client/my-sites/site-settings/speed-up-site-settings.jsx

```diff
@@ -92,18 +169,76 @@ class SpeedUpSiteSettings extends Component {
 	}
 }
 
-export default connect( state => {
-	const selectedSiteId = getSelectedSiteId( state );
-	const siteInDevMode = isJetpackSiteInDevelopmentMode( state, selectedSiteId );
-	const moduleUnavailableInDevMode = isJetpackModuleUnavailableInDevelopmentMode(
-		state,
-		selectedSiteId,
-		'photon'
-	);
+export default connect(
+	state => {
+		const selectedSiteId = getSelectedSiteId( state );
+		const siteInDevMode = isJetpackSiteInDevelopmentMode( state, selectedSiteId );
+		const siteAcceleratorSupported = isJetpackMinimumVersion( state, selectedSiteId, '6.6-alpha' );
+		const moduleUnavailableInDevMode = isJetpackModuleUnavailableInDevelopmentMode(
+			state,
+			selectedSiteId,
+			'photon'
+		);
+		const photonModuleActive = isJetpackModuleActive( state, selectedSiteId, 'photon' );
```

Comment:
FWIW we already have access to those for free - the `settings` prop that is passed from `wrapSettingsForm` contains a list of all module settings, but also of the activation state of each module. I wonder if using that would simplify our code here...

Reply:
I must be doing something wrong, I can't seem to get this to work. Here is what I've tried:

```diff
diff --git a/client/my-sites/site-settings/form-writing.jsx b/client/my-sites/site-settings/form-writing.jsx
index aa429155a9..ceae5cfccb 100644
--- a/client/my-sites/site-settings/form-writing.jsx
+++ b/client/my-sites/site-settings/form-writing.jsx
@@ -149,6 +149,7 @@ class SiteSettingsFormWriting extends Component {
 								jetpackVersionSupportsLazyImages={ jetpackVersionSupportsLazyImages }
 								submitForm={ submitForm }
 								updateFields={ updateFields }
+								fields={ fields }
 							/>
 						</div>
 					) }
diff --git a/client/my-sites/site-settings/speed-up-site-settings.jsx b/client/my-sites/site-settings/speed-up-site-settings.jsx
index 6f286ad089..53563d8071 100644
--- a/client/my-sites/site-settings/speed-up-site-settings.jsx
+++ b/client/my-sites/site-settings/speed-up-site-settings.jsx
@@ -16,14 +16,18 @@ import CompactFormToggle from 'components/forms/form-toggle/compact';
 import FormFieldset from 'components/forms/form-fieldset';
 import { getSelectedSiteId } from 'state/ui/selectors';
 import { getSiteSlug, isJetpackMinimumVersion } from 'state/sites/selectors';
-import isJetpackModuleActive from 'state/selectors/is-jetpack-module-active';
 import isJetpackModuleUnavailableInDevelopmentMode from 'state/selectors/is-jetpack-module-unavailable-in-development-mode';
 import isJetpackSiteInDevelopmentMode from 'state/selectors/is-jetpack-site-in-development-mode';
 import JetpackModuleToggle from 'my-sites/site-settings/jetpack-module-toggle';
 import SupportInfo from 'components/support-info';
 
 class SpeedUpSiteSettings extends Component {
+	static defaultProps = {
+		fields: {},
+	};
+
 	static propTypes = {
+		fields: PropTypes.object,
 		isRequestingSettings: PropTypes.bool,
 		isSavingSettings: PropTypes.bool,
 		jetpackVersionSupportsLazyImages: PropTypes.bool,
@@ -31,17 +35,18 @@ class SpeedUpSiteSettings extends Component {
 		updateFields: PropTypes.func.isRequired,
 
 		// Connected props
-		assetCdnModuleActive: PropTypes.bool,
-		photonModuleActive: PropTypes.bool,
 		photonModuleUnavailable: PropTypes.bool,
 		selectedSiteId: PropTypes.number,
-		siteAcceleratorStatus: PropTypes.bool,
 		siteAcceleratorSupported: PropTypes.bool,
 		siteSlug: PropTypes.string,
 	};
 
 	handleCdnChange = () => {
-		const { siteAcceleratorStatus, submitForm, updateFields } = this.props;
+		const { fields, submitForm, updateFields } = this.props;
+
+		// Status of the main site accelerator toggle.
+		const { photon: photonModuleActive, "photon-cdn": assetCdnModuleActive } = fields;
+		const siteAcceleratorStatus = !! ( photonModuleActive || assetCdnModuleActive );
 
 		// If one of them is on, we turn everything off.
 		updateFields(
@@ -57,17 +62,21 @@ class SpeedUpSiteSettings extends Component {
 
 	render() {
 		const {
+			fields,
 			isRequestingSettings,
 			isSavingSettings,
 			jetpackVersionSupportsLazyImages,
 			photonModuleUnavailable,
 			selectedSiteId,
 			siteAcceleratorSupported,
-			siteAcceleratorStatus,
 			translate,
 		} = this.props;
 		const isRequestingOrSaving = isRequestingSettings || isSavingSettings;
 
+		// Status of the main site accelerator toggle.
+		const { photon: photonModuleActive, "photon-cdn": assetCdnModuleActive } = fields;
+		const siteAcceleratorStatus = !! ( photonModuleActive || assetCdnModuleActive );
+
 		return (
 			<div className="site-settings__module-settings site-settings__speed-up-site-settings">
 				<Card>
@@ -147,18 +156,10 @@ export default connect( state => {
 		selectedSiteId,
 		'photon'
 	);
-	const photonModuleActive = isJetpackModuleActive( state, selectedSiteId, 'photon' );
-	const assetCdnModuleActive = isJetpackModuleActive( state, selectedSiteId, 'photon-cdn' );
-
-	// Status of the main site accelerator toggle.
-	const siteAcceleratorStatus = !! ( photonModuleActive || assetCdnModuleActive );
 
 	return {
-		assetCdnModuleActive,
-		photonModuleActive,
 		photonModuleUnavailable: siteInDevMode && moduleUnavailableInDevMode,
 		selectedSiteId,
-		siteAcceleratorStatus,
 		siteAcceleratorSupported,
 		siteSlug: getSiteSlug( state, selectedSiteId ),
 	};

```

Reply:
What you're doing makes sense actually - I haven't tried it, but I'm surprised it doesn't work.

I'm actually fine with the current approach too - let's not spend more time further polishing it, if it's already good 👍 .

Let me know when this one is ready for another look.

Reply:
👍 	
This should be ready for testing then.